### DISCUSSION TITLE
Getting error when i try to plot the geometry

### DISCUSSION THREAD

**Discussion by Indrajeetrt** [ansys/pyansys-geometry#1384] [Q&A]
Hello,
I am trying to use these line of code to read a geometry and plot it. 3 weeks back this GeometryPlotter() worked for me but when i try now it is giving me error saying **GeometryExitedError: Geometry service connection terminated: The given key was not present in the dictionary.** it would be helpfull if anyone has some clues about the error..
Thank you in advance
**from ansys.geometry.core import launch_modeler_with_spaceclaim
from ansys.geometry.core.plotting import GeometryPlotter
#Lunch modeler with spaceclaim and 
modeler = launch_modeler_with_spaceclaim(hidden=True)
geom = modeler.open_file(fileName,False)
plotter = GeometryPlotter()
plotter.show(geom)**

**Comment by RobPasMue** [comment]
Hi @Indrajeetrt - transfering this discussion to the PyAnsys Geometry repository!

**Comment by RobPasMue** [comment]
Hi @Indrajeetrt - I gave it a try to this code on my side (I just defined a specific file to be uploaded, which I am also providing here):

File used: https://github.com/ansys/pyansys-geometry/blob/main/tests/integration/files/import/box.3dm

```py
from pathlib import Path

from ansys.geometry.core import launch_modeler_with_spaceclaim
from ansys.geometry.core.plotting import GeometryPlotter

fileName = Path.cwd() / 'tests' / 'integration' / 'files' / 'import' / 'box.3dm'

# Launch modeler with spaceclaim
modeler = launch_modeler_with_spaceclaim(hidden=True)
geom = modeler.open_file(fileName.as_posix(), upload_to_server=False)

plotter = GeometryPlotter()
plotter.show(geom)

modeler.close()
```

Question...

* What SpaceClaim version are you using?
* What PyAnsys Geometry version are you using?
* How are you passing in the file name? It should be a string. If you are passing a ``pathlib.Path`` object, you need to do just like I did --> ``path.as_posix()``

I will adapt this last point to also accept Path objects but this will have to be done in a new release =)

Also, just for convenience... You don't need to instantiate a ``GeometryPlotter`` especifically if you just want to visualize the entire geometry or a body. You can also do the following:

```py
geom = modeler.open_file(fileName.as_posix(), upload_to_server=False)
geom.plot()

modeler.close()
```

**Reply by Indrajeetrt** [reply]
1. I have 2024R1 spaceclaim installed in my system
2. 0.6.6 is my pyansys geometry
3. I am passing the filename as string and it's a step file

**Reply by RobPasMue** [reply]
Hi @Indrajeetrt - I adapted my code to use a ``stp`` file, PyAnsys Geometry version 0.6.6 and SpaceClaim version 24R1. You can find the file I used here: https://github.com/ansys/pyansys-geometry/blob/main/tests/integration/files/import/twoCars.stp

```py
from pathlib import Path

from ansys.geometry.core import launch_modeler_with_spaceclaim, __version__
from ansys.geometry.core.plotting import GeometryPlotter

fileName = Path.cwd() / 'tests' / 'integration' / 'files' / 'import' / 'twoCars.stp'

print(f"Running with version {__version__}")

# Launch modeler with spaceclaim
modeler = launch_modeler_with_spaceclaim(hidden=True, product_version=241)
geom = modeler.open_file(fileName.as_posix(), upload_to_server=False)

plotter = GeometryPlotter()
plotter.show(geom)

modeler.close()
```

See my screenshot below demonstrating it runs.

![image](https://github.com/user-attachments/assets/7c6d3309-bd58-4159-8bc2-c70946407939)

Could you give it a try to this same code and file? Also, would it be possible for you to share the ``.stp`` file you are using?

**Reply by Indrajeetrt** [reply]
I have a step file of around 80 mb which I am not able to visualise.. but I have a step file of size 3 mb which I am able to to visualise.. Is there any restriction for size of the step file ?

**Reply by RobPasMue** [reply]
There is no restriction in size as far as I know. Do you get any error or does it just stale?

**Reply by Indrajeetrt** [reply]
I get this below error..
"GeometryExitedError: Geometry service connection terminated: The given key was not present in the dictionary"

**Reply by RobPasMue** [reply]
Is it the exact same PyAnsys Geometry code and just changing the file?

**Reply by RobPasMue** [reply]
If possible, would you be able to share the step files and the code?

**Reply by Indrajeetrt** [reply]
Code is same.. But I can't share the step file..

**Reply by RobPasMue** [reply]
Okay... I have some intuition. Can you explicitly change this line on your code

```py
geom = modeler.open_file(fileName,False)
```

to

```py
geom = modeler.open_file(fileName,upload_to_server=False)
```

And run it again?

**Reply by RobPasMue** [reply]
Or is the code failing whenever you perform the call to ``plotter.show()``? Can you share the entire stack trace?

**Reply by Indrajeetrt** [reply]
Yes true, it is giving error when I perform plotter.show()

**Reply by RobPasMue** [reply]
Can you share the entire stack trace of the error? I'd like to see which specific call is the one causing the error because the plotter.show() method has a lot of underlying calls.

**Reply by Indrajeetrt** [reply]
Here is entire stack trace of the error
[error.txt](https://github.com/user-attachments/files/16860540/error.txt)

**Reply by RobPasMue** [reply]
So there seems to be some issues with the way the model is read by PyAnsys Geometry and the way the model is loaded by SpaceClaim. Error is occurring when the tessellation is requested on your model.

My next step to identify the problem would be to make sure that the model you have on the client reflects the model available on SpaceClaim.

I would start accessing: ``design.components``, ``design.bodies`` and in every component, checking its subcomponents... I would also recommend you put a debug point on your ``lib\site-packages\ansys\geometry\core\designer\body.py:1408`` line and check the body.name of it before failing. You can even do it with a print statement if you prefer:

```python
    ) -> Union["PolyData", "MultiBlock"]:
        print(self.name)
        return self._template.tessellate(merge, self.parent_component.get_world_transform())
```

This would let us know what body is the one causing problems. Once identified, I would simplify your step file to only include that specific body and check whether it still fails or not. If possible I would also ask for you to share the simplified step file with us (if you consider that just that element alone is fine to share).

Bottom line - my feeling is that there might be some issue with your specific model and the way things are read by the service, rather than a consistent problem with any step file... let's try to debug it together! 😄
